src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+
+#pragma comment(lib, "d3d11.lib")
+#pragma comment(lib, "dxgi.lib")
+#pragma comment(lib, "d3dcompiler.lib")
+#pragma comment(lib, "winmm.lib")
+#pragma comment(lib, "dxguid.lib")
+
+struct VertexFormat
+{
+    float vertex[3];
+    float color[3];
+};
+
+HelloCOMApplication::HelloCOMApplication(const std::string_view title)
+    : Application(title)
+{
+}
+
+HelloCOMApplication::~HelloCOMApplication()
+{
+    _deviceContext->Flush();
+    _pixelShader.Reset();
+    _vertexShader.Reset();
+    _vertexLayout.Reset();
+    _triangleIndices.Reset();
+    _triangleVerts.Reset();
+    _renderTarget.Reset();
+    _swapChain.Reset();
+    _dxgiFactory.Reset();
+#if !defined(NDEBUG)
+    _debug->ReportLiveDeviceObjects(D3D11_RLDO_SUMMARY);
+    _debug.Reset();
+#endif
+    _deviceContext.Reset();
+    _device.Reset();
+    Application::Cleanup();
+}
+
+bool HelloCOMApplication::Initialize()
+{
+    // This section initializes GLFW and creates a Window
+    if (!Application::Initialize())
+    {
+        return false;
+    }
+    glfwSetWindowUserPointer(GetWindow(), this);
+    glfwSetFramebufferSizeCallback(GetWindow(), OnResize);
+    const HWND nativeWindow = glfwGetWin32Window(GetWindow());
+    // This section initializes DirectX's devices and SwapChain
+    if (FAILED(CreateDXGIFactory1(__uuidof(IDXGIFactory1), &_dxgiFactory)))
+    {
+        std::cout << "DXGI: Unable to create DXGIFactory\n";
+        return false;
+    }
+    _dxgiFactory->MakeWindowAssociation(nativeWindow, 0);
+    constexpr D3D_FEATURE_LEVEL deviceFeatureLevel = D3D_FEATURE_LEVEL_11_0;
+    UINT deviceFlags = D3D11_CREATE_DEVICE_BGRA_SUPPORT;
+#if !defined(NDEBUG)
+    deviceFlags |= D3D11_CREATE_DEVICE_DEBUG;
+#endif
+    if (FAILED(D3D11CreateDevice(
+        nullptr,
+        D3D_DRIVER_TYPE_HARDWARE,
+        nullptr,
+        deviceFlags,
+        &deviceFeatureLevel,
+        1,
+        D3D11_SDK_VERSION,
+        &_device,
+        nullptr,
+        &_deviceContext)))
+    {
+        std::cout << "D3D11: Failed to create Device and Device Context\n";
+        return false;
+    }
+#if !defined(NDEBUG)
+    if (FAILED(_device.As(&_debug)))
+    {
+        std::cout << "D3D11: Failed to get Device Debug Context\n";
+        return false;
+    }
+    if (FAILED(_debug.As(&_debugInfoQueue)))
+    {
+        std::cout << "D3D11: Failed to get Debug Info Queue\n";
+        return false;
+    }
+    _debugInfoQueue->SetBreakOnSeverity(D3D11_MESSAGE_SEVERITY_CORRUPTION, true);
+    _debugInfoQueue->SetBreakOnSeverity(D3D11_MESSAGE_SEVERITY_ERROR, true);
+
+    D3D11_MESSAGE_ID hide[] =
+    {
+        D3D11_MESSAGE_ID_SETPRIVATEDATA_CHANGINGPARAMS,
+        // Add more message IDs here as needed
+    };
+
+    D3D11_INFO_QUEUE_FILTER filter = {};
+    filter.DenyList.NumIDs = _countof(hide);
+    filter.DenyList.pIDList = hide;
+    _debugInfoQueue->AddStorageFilterEntries(&filter);
+    _debugInfoQueue.Reset();
+#endif
+
+    DXGI_SWAP_CHAIN_DESC swapchainInfo;
```

Comment:
Hmmmm I also think we should init these descriptors with `= {};` just in case, probably a good exercise anyway to do it that way